edflow/custom_logging.py

```diff
@@ -423,21 +423,63 @@ def _create_logger(name, out_dir, level=logging.INFO):
         logger = logging.getLogger(name)
         logger.setLevel(level)
 
+        logger.copyline = lambda msg: logger.log(22, msg)
+
         if not len(logger.handlers) > 0:
             ch = TqdmHandler()
             fh = logging.FileHandler(filename=os.path.join(out_dir, "log.txt"))
 
             fmt_string = "[%(levelname)s] [%(name)s]: %(message)s"
             formatter = logging.Formatter(fmt_string)
-            fh.setFormatter(formatter)
-            ch.setFormatter(formatter)
+            fh.setFormatter(CopyLineFormatter(False))
+            ch.setFormatter(CopyLineFormatter(True))
 
             logger.addHandler(ch)
             logger.addHandler(fh)
 
         return logger
 
 
+class CopyLineFormatter(logging.Formatter):
+    def __init__(self, use_color, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.BLACK, self.RED, self.GREEN, self.YELLOW = range(4)
+        self.BLUE, self.MAGENTA, self.CYAN, self.WHITE = range(4, 8)
+
+        # The background is set with 40 plus the number of the color, and the
+        # foreground with 30
+
+        # These are the sequences need to get colored ouput
+        self.RESET_SEQ = "\033[0m"
+        self.COLOR_SEQ = "\033[{}m"
+        self.BOLD_SEQ = "\033[1m"
+
+        if use_color:
+            self._fmt_str = "{color}[{levelname}] [{name}]: {msg}{color_end}"
+        else:
+            self._fmt_str = "[{levelname}] [{name}]: {msg}"
+
+    def format(self, record):
+
+        content = {
+            "color": "",
+            "color_end": "",
+            "levelname": record.levelname,
+            "name": record.name,
+            "msg": record.msg,
+        }
+
+        if record.levelno == 22:
```

Comment:
can we make this formatter generic enough so that we can also do something like `colored_info("red", msg)` for custom colored logging output?

Reply:
I like the idea! I will have a look at it.

Reply:
Added in my latest [commit](https://github.com/pesser/edflow/pull/230/commits/168b3725c04104ba16033d84edb47f18705c9add)